src/pip/_internal/cli/main_parser.py

```diff
@@ -47,6 +49,44 @@ def create_main_parser() -> ConfigOptionParser:
     return parser
 
 
+def identify_python_interpreter(python: str) -> Optional[str]:
+    if python == "python" or python == "py":
+        # Run the active Python.
+        # We have to be very careful here, because:
+        #
+        # 1. On Unix, "python" is probably correct but there is a "py" launcher.
+        # 2. On Windows, "py" is the best option if it's present.
+        # 3. On Windows without "py", "python" might work, but it might also
+        #    be the shim that launches the Windows store to allow you to install
+        #    Python.
+        #
+        # We go with getting py on Windows, and if it's not present or we're
+        # on Unix, get python. We don't worry about the launcher on Unix or
+        # the installer stub on Windows.
+        py = None
+        if WINDOWS:
+            py = shutil.which("py")
+        if py is None:
+            py = shutil.which("python")
+        if py:
+            return py
+
+    # TODO: On Windows, `--python .venv/Scripts/python` won't pass the
+    #       exists() check (no .exe extension supplied). But it's pretty
+    #       obvious what the user intends. Should we allow this?
```

Comment:
We could do this without too much fuss with

```python
directory, filename = os.path.split(python)
if shutil.which(filename, path=directory) is not None:
    ...
```

Reply:
From the `shutil.which` docs,

 > On Windows, the current directory is always prepended to the path whether or not you use the default or provide your own

That could give weird false positives (for example, we have a `python.exe` in the CWD, and we do `--python=.venv/python.exe`).

On reflection, I think this is too much effort, and we should just expect the user to supply the `.exe` (i.e., provide the name of a file that exists). I suspect most people will use autocomplete anyway, if they are entering the path by hand.